admin/validate_horace.m

```diff
@@ -214,38 +222,53 @@
 
 end
 
-function test_stage_reset(hoc, hpc, par, nomex, forcemex, talkative)
+
+%-------------------------------------------------------------------------------
+function test_stage_reset(icount, hor, hpc, par, nomex, forcemex, talkative)
 % Run before each stage
 % Set Horace configurations to the defaults (but don't save)
 % (The validation should be done starting with the defaults, otherwise an error
 %  may be due to a poor choice by the user of configuration parameters)
 
-    set(hoc, 'defaults');
-    set(hpc, 'defaults');
-    % set(par, 'defaults');
+% Set the default configurations, printing warning only the first time round to
+% avoid copious warning messages
+warn_state = warning();
+cleanup_obj = onCleanup(@()warning(warn_state));
+if icount>1
+    warning('off',  'all');
+end
```

Comment:
When running on my laptop, I was getting warning messages for each test folder in a call validate_horace, which means c. 40 repeats of the same set of warning messages thereby dominating the output. I put the the cleanup in the function so that safe recovery of the incoming warning state on error is guaranteed, without having to worry about what recoding might be done elsewhere in validate_horace at a later date. Losing the warning messages for the remainder of a Matlab session would be bad!